### PR TITLE
enable eslint and fix code issues

### PR DESCRIPTION
**Pull request by davidchaiken**
* configured eslint to run on nodejs code
* fixed issues from eslint automatically and manually
* added eslint to github actions workflow
* documented eslint-based code conventions in README and CONTRIBUTING files

### CODE REVIEW

File: nodejs/src/access_token_common.js [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
+  constructor(api_config, { name }) {
     super(api_config);
 
     const auth = `${api_config.app_id}:${api_config.app_secret}`;
     const b64auth = Buffer.from(auth).toString('base64');
     this.api_uri = api_config.api_uri;
-    this.auth_headers = {Authorization: `Basic ${b64auth}`}
-    this.name = (name) ? name : `access_token_${api_config.version}`;
+    this.auth_headers = { Authorization: `Basic ${b64auth}` };
+    this.name = (name) || `access_token_${api_config.version}`;
```
Comment (thisjuvy) [inline nodejs/src/access_token_common.js:15]: nit: you can remove the () e.g name || `access_token_${api_config.version}`;

File: nodejs/src/user_auth.js [hunk 2]
```diff
@@ -10,42 +9,41 @@ import open from 'open'
...
     // Capture the open sockets so that the server can be terminated completely.
-    server.on('connection', function (socket) { sockets.push(socket); });
+    server.on('connection', function(socket) { sockets.push(socket); });
 
     // Start listening on the requested port.
     server.listen(api_config.port);
   });
 
-  var access_uri = (api_config.oauth_uri + '/oauth/' +
+  let access_uri = (api_config.oauth_uri + '/oauth/' +
```
Comment (thisjuvy) [inline nodejs/src/user_auth.js:46]: do we want to strictly adopt template strings?

File: nodejs/src/access_token_common.js (hunk 1 above)
Comment (davidchaiken) [reply nodejs/src/access_token_common.js:15]: Good point. Let me see if there's an eslint rule that will check for extra parentheses.

File: nodejs/src/utils.js [hunk 3]
```diff
@@ -49,7 +49,7 @@ export class Input {
       if (value === '') {
         return defaultValue; // blank entry returns the default
       }
-      for (var idx = 0; idx < one_of_list.length; ++idx) {
+      for (let idx = 0; idx < one_of_list.length; ++idx) {
```
Comment (thisjuvy) [inline nodejs/src/utils.js:52]: perfect usecase for filter 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

File: nodejs/src/user_auth.js (hunk 2 above)
Comment (davidchaiken) [reply nodejs/src/user_auth.js:46]: I adopted template strings in almost all of the other instances. In this case, I decided to split a very long strong over multiple lines using simple concatenation. Is there a preferred JS idiom for splitting template strings over multiple lines?

File: nodejs/src/utils.js (hunk 3 above)
Comment (davidchaiken) [reply nodejs/src/utils.js:52]: Good point. I'd prefer to restrict this PR to eslint-specific changes.

File: nodejs/src/user_auth.js (hunk 2 above)
Comment (davidchaiken) [reply nodejs/src/user_auth.js]: ...and I'd prefer for this PR to focus on eslint-related changes.

File: nodejs/src/access_token_common.js (hunk 1 above)
Comment (davidchaiken) [reply nodejs/src/access_token_common.js]: Added, autofixed, and pushed the eslint no-extra-parens rule.

**Review by thisjuvy — APPROVED** [review]

**Pull request merged**